source/extensions/tracers/skywalking/skywalking_types.cc

```diff
@@ -0,0 +1,174 @@
+#include "extensions/tracers/skywalking/skywalking_types.h"
+
+#include "envoy/common/exception.h"
+
+#include "common/common/base64.h"
+#include "common/common/empty_string.h"
+#include "common/common/fmt.h"
+#include "common/common/hex.h"
+#include "common/common/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace SkyWalking {
+
+namespace {
+
+// The standard header name is "sw8", as mentioned in:
+// https://github.com/apache/skywalking/blob/v8.1.0/docs/en/protocols/Skywalking-Cross-Process-Propagation-Headers-Protocol-v3.md.
+const Http::LowerCaseString& propagationHeader() {
+  CONSTRUCT_ON_FIRST_USE(Http::LowerCaseString, "sw8");
+}
+
+std::string generateId(Random::RandomGenerator& random_generator) {
+  return absl::StrCat(Hex::uint64ToHex(random_generator.random()),
+                      Hex::uint64ToHex(random_generator.random()));
+}
+
+std::string base64Encode(const absl::string_view input) {
+  return Base64::encode(input.data(), input.length());
+}
+
+// Decode and validate fields of propagation header.
+std::string base64Decode(absl::string_view input) {
+  std::string result = input.length() % 4 ? EMPTY_STRING : Base64::decodeWithoutPadding(input);
+  if (result.empty()) {
+    throw EnvoyException("Invalid propagation header for SkyWalking: parse error");
+  }
+  return result;
+}
+
+} // namespace
+
+SpanContextPtr SpanContext::spanContextFromRequest(Http::RequestHeaderMap& headers) {
+  auto propagation_header = headers.get(propagationHeader());
+  if (propagation_header.empty()) {
+    // No propagation header then Envoy is first hop.
+    return nullptr;
+  }
+
+  auto header_value_string = propagation_header[0]->value().getStringView();
+  const auto parts = StringUtil::splitToken(header_value_string, "-", false, true);
+  // Reference:
+  // https://github.com/apache/skywalking/blob/v8.1.0/docs/en/protocols/Skywalking-Cross-Process-Propagation-Headers-Protocol-v3.md.
+  if (parts.size() != 8) {
+    throw EnvoyException(
+        fmt::format("Invalid propagation header for SkyWalking: {}", header_value_string));
+  }
```

Comment:
nit: Probably, we should add documents that Envoy Skywalking tracer has no capability to accept `sw8-x` header.